provider/cluster/manager.go

```diff
@@ -312,6 +314,14 @@ func (dm *deploymentManager) doDeploy() ([]string, error) {
 		case <-ctx.Done():
 		}
 	}()
+
+	if err = dm.checkLeaseActive(ctx); err != nil {
```

Comment:
added active lease guard here @troian @hydrogen18

Reply:
I took out the previous guard (`ActiveLeasesForProvider`) so I wanted to replace it with something here.  I accidentally launched 15,000 namespaces without it (didn't have manifest pruning).

Even with namespace pruning, offhand I can think of:

1. provider wins lease
2. provider receives manifest
3. provider shuts down
4. lease is closed by client
5. provider starts up

I agree that a vacuum process (handled by `orphan-ns-purge` script now) does make sense, but I figure it's not too much to do a query before creating a ns just in case as well.

Reply:
Now I'm second guessing this - maybe it should just be a re-implementation of `ActiveLeasesForProvider`.  Worried about a torrent of parallel queries on startup.  At least the retry has jitter? :man_shrugging:

Reply:
Is this due to the lack of a secondary index? Can we just query deployment first (by deploymentID) and if it is closed then we're done?

Reply:
Yeah, lack of secondary index made [`ActiveLeasesForProvider()`](https://github.com/ovrclk/akash/blob/13abbb6c4d88d93bf95c4b313fecf9818c0e336c/client/extra.go#L13-L30)  unuseable.  We can actually just query the lease by id, which is what we do in this PR.

Reply:
I don't think the place of this check really matters at this time. Also, in this change you've moved it to its own goroutine (from what I can see) so this should be fine.

Reply:
So at this point it seems we should really add a secondary index as part of the next upgrade? I'm not seeing an easy workaround to this presently.

Reply:
I think it's too late to add a secondary index, and I'm hesitant to in the first place.

* this is covering a corner-case - lease open but no manifest yet when restarting provider
* it can be solved in other ways
  * manifest server can lookup leases when a manifest for an unknown lease is received
  * active leases can be cached locally in etcd or similar

I might actually be ok with removing this second `ActiveLeasesForProvider()` invocation completely and living with the corner case not being handled for the time being.

On the other hand, having only one invocation of this seems to fix the problem in our larger clusters.

Reply:
I suggest we go ahead and remove it. It's a corner case at this time.